core/src/main/java/org/elasticsearch/index/query/support/InnerHitBuilder.java

```diff
@@ -202,7 +200,7 @@ public String getName() {
     }
 
     public InnerHitBuilder setName(String name) {
-        this.name = name;
+        this.name = Objects.requireNonNull(name);
```

Comment:
Is it valid for an inner hits builder to have no name, nestedPath and/or parentChildType set? If not then maybe we should remove the setters for these three fields and have them instead set in the constructor so that it is not possible to construct an InnerHitsBuilder that is not valid?